File: sujets/elodie-sevestre/exercice-b.js
// Exercice B — Débogage Express + PostgreSQL
//
// Voici un handler POST /competences qui doit créer une compétence
// rattachée à un projet (stack Express + PostgreSQL via le driver `pg`).
//
// Ce code contient 4 bugs. Repère-les et corrige-les directement ici.
// Lis le code à voix haute et explique ce qui cloche au fur et à mesure.

const pool = require("./db"); // pool pg déjà configuré

// POST /competences — crée une compétence rattachée à un projet
app.post("/competences", async (req, res) => {
  const { nom, categorie, projet_id } = req.params;
  try {
    const result = await pool.query(
      "INSERT INTO competences (nom, categorie, projet_id) VALUES (?, ?, ?)",
      [nom, categorie, projet_id]
    );
    res.status(201).json(result.rows[0]);
  } catch (err) {
    res.json({ error: err.message });
  }
});

File: sujets/mariama-bah/ExerciceB.cs
// Exercice B — Débogage ASP.NET Core + EF Core
//
// Voici un endpoint POST /conges qui doit enregistrer une demande de congé
// d'un employé (stack ASP.NET Core minimal API + Entity Framework Core).
//
// Ce code contient 4 bugs. Repère-les et corrige-les directement ici.
// Lis le code à voix haute et explique ce qui cloche au fur et à mesure.

public class Employe
{
    public int Id { get; set; }
    public string Nom { get; set; } = "";
}

public class Conge
{
    public int Id { get; set; }
    public int EmployeId { get; set; }
    public DateOnly Debut { get; set; }
    public DateOnly Fin { get; set; }
}

// PlanningDb est un DbContext EF Core déjà configuré,
// avec DbSet<Employe> Employes et DbSet<Conge> Conges.

// POST /conges — enregistrer une demande de congé d'un employé
app.MapPost("/conges", (PlanningDb db, int employeId, DateOnly debut, DateOnly fin) =>
{
    var employe = db.Employes.Find(employeId);
    var conge = new Conge { EmployeId = employe.Id, Debut = debut, Fin = fin };

    db.Conges.Add(conge);

    return Results.Ok(conge);
});
